scripts/rebuild-filterlist.mjs

```diff
@@ -3,14 +3,179 @@
 import fs from 'fs';
 import path from 'path';
 import { fileURLToPath } from 'url';
+import { parse } from 'csv-parse/sync';
+import { getDomain } from 'tldts-experimental';
+import { Readable } from 'stream';
+import { finished } from 'stream/promises';
 
 const rulesDir = path.join(path.dirname(fileURLToPath(import.meta.url)), '../rules');
+const dataDir = path.join(path.dirname(fileURLToPath(import.meta.url)), '../data');
 
 const easylistRevision = fs.readFileSync(path.join(rulesDir, 'filterlists', 'easylist_revision.txt'), 'utf-8');
 
-// TODO: consider using python-abp (flrender) to generate filterlist properly
-const filterlistContent = `
-[Adblock Plus 2.0]
+const MAX_DOMAIN_RANK = 100000;
+
+/** @typedef {{ rules: Record<string, string[]>}} FilterlistJSON */
+
+/** @type {Map<string, string>} */
+let domainMap;
+
+/**
+ * Processes a list, filters out unsupported rules, and removes longtail domain rules
+ * @param {String} listFileName
+ */
+async function processFilterList(listFileName) {
+    console.log(`Processing ${listFileName}`);
+    const data = fs.readFileSync(path.join(rulesDir, 'filterlists', listFileName), 'utf-8');
+    const lines = data.split('\n');
+    /** @type {FilterlistJSON} */
+    const filterlistJSON = { rules: {} };
+
+    // Remove unsupported rule types
+    const filteredLines = lines.filter(
+        (line) =>
+            !line.startsWith('!') &&
+            !line.startsWith('##') &&
+            !line.startsWith('||') &&
+            !line.includes(':remove') &&
+            !line.includes(':upward') &&
+            !line.includes('redirect-rule'),
```

Comment:
Could you add some comments on what each of these prefixes mean and why we're filtering them out?